tests/test_private_vlan.rb

```diff
@@ -0,0 +1,777 @@
+# limitations under the License.
+
+require_relative 'ciscotest'
+require_relative '../lib/cisco_node_utils/vlan'
+
+include Cisco
+
+# TestVlan - Minitest for Vlan node utility
+class TestVlan < CiscoTestCase
+  @skip_unless_supported = 'vlan'
+
+  @@cleaned = false # rubocop:disable Style/ClassVars
+  def cleanup
+    Vlan.vlans.each do |vlan, obj|
+      # skip reserved vlans
+      next if vlan == '1'
+      next if node.product_id[/N5K|N6K|N7K/] && (1002..1005).include?(vlan.to_i)
+      obj.destroy
+    end
+    config('no feature vtp')
+    config('no feature private-vlan')
+  end
+
+  def setup
+    super
+    cleanup unless @@cleaned
+    @@cleaned = true # rubocop:disable Style/ClassVars
+  end
+
+  def teardown
+    cleanup
+    config('no feature private-vlan')
+    config('no feature vtp')
+  end
+
+  def test_private_type_default
+    config('no feature vtp')
+    config('feature private-vlan')
+    v1 = Vlan.new(100)
+    pv_type = ''
+    if validate_property_excluded?('vlan', 'private_vlan_type')
+      assert_nil(v1.private_vlan_type)
+    else
+      assert_equal(pv_type, v1.private_vlan_type)
+    end
+    v1.destroy
+  end
+
+  def test_private_association_default
+    config('no feature vtp')
+    config('feature private-vlan')
+    v1 = Vlan.new(100)
+    if validate_property_excluded?('vlan', 'private_vlan_type')
+      assert_nil(v1.private_vlan_type)
+      v1.destroy
+      return
+    else
+      result = ''
+      assert_equal(result, v1.private_vlan_association)
+    end
+  end
+
+  def test_private_vlan_type_primary
+    v1 = Vlan.new(100)
+    pv_type = 'primary'
+    if validate_property_excluded?('vlan', 'private_vlan_type')
+      assert_nil(v1.private_vlan_type)
+      assert_raises(Cisco::UnsupportedError) do
+        v1.private_vlan_type = pv_type
+      end
+    else
+      v1.private_vlan_type = pv_type
+      assert_equal(pv_type, v1.private_vlan_type)
+    end
+    v1.destroy
+  end
+
+  def test_no_private_vlan_type_primary
+    v1 = Vlan.new(200)
+    pv_type = 'primary'
+    if validate_property_excluded?('vlan', 'private_vlan_type')
+      assert_nil(v1.private_vlan_type)
+      assert_raises(Cisco::UnsupportedError) do
+        v1.private_vlan_type = pv_type
+      end
+      v1.destroy
+      return
+    else
+      v1.private_vlan_type = pv_type
+      assert_equal(pv_type, v1.private_vlan_type)
+      pv_type = ''
+      v1.private_vlan_type = pv_type
+      assert_equal(pv_type, v1.private_vlan_type)
+      v1.destroy
+    end
+  end
+
+  def test_multi_private_vlan_type_primary
+    v1 = Vlan.new(100)
+    v2 = Vlan.new(101)
+    v3 = Vlan.new(200)
+    v4 = Vlan.new(201)
+    v5 = Vlan.new(203)
+    pv_type = 'primary'
+    if validate_property_excluded?('vlan', 'private_vlan_type')
+      assert_nil(v1.private_vlan_type)
+      assert_raises(Cisco::UnsupportedError) do
+        v1.private_vlan_type = pv_type
+      end
+      v1.destroy
```

Comment:
Given that you are disabling the features in your `teardown` method, you don't need all of these `destroy` calls right?